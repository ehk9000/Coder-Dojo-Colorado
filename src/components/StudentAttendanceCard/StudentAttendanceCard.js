import React, { useState } from 'react';

function StudentAttendanceCard({ student }) {
  const [guardianInfo, showGuardianInfo] = useState(false);
  const { 
    name, username, email, phoneNumber 
  } = student.guardianId;
  const guardianInfoBox = (
    <div className="guardian-details">
      <p>Guardian Details:</p>
      <p>
        {name}
        :
        {username}
      </p>
      <p>
        Phone:
        {phoneNumber}
      </p>
      <p>
        Email:
        {email}
      </p>
    </div>
  );
  return (
    <article 
      onMouseEnter={() => showGuardianInfo(true)} 
      onMouseLeave={() => showGuardianInfo(false)} 
      key={student.id}
    >
      <h4>
        {student.name}
        :
        {student.username}
      </h4>
      {guardianInfo && guardianInfoBox}
    </article>
  );
}

export default StudentAttendanceCard;
