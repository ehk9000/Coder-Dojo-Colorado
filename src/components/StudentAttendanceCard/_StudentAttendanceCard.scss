.StudentAttendanceCard {
  margin: 10px;
  position: relative;

  .guardian-details {
    position: absolute;
    background-color: #F4F5F6;
    color: #724695;
    font-size: .8rem;
    z-index: 2;
    padding: 10px;
  }
}