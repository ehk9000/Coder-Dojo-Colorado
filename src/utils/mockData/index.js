/* eslint-disable */
const mockData = {
  mockAdminData: {
    "me": {
        "email": "example@admin1.com",
        "phoneNumber": "5555555553",
        "username": "admin1",
        "addresses": [
            {
                "street1": "234 Main St.",
                "street2": "Apt 456",
                "city": "denver",
                "state": "CO",
                "zip": "80128"
            }
        ],
        "venues": [
            {
                "id": 1,
                "name": "Venue One",
                "notes": "Here be Dragons",
                "email": "example@venue.com",
                "webUrl": "www.venueone.com",
                "events": [
                    {
                        "id": 2,
                        "name": "Test Event 2",
                        "notes": "testing notes",
                        "startTime": "2019-08-21 17:00:00 UTC",
                        "endTime": "2019-08-21 18:00:00 UTC",
                        "eventCode": "45fbf7"
                    },
                    {
                        "id": 3,
                        "name": "Test Event 3",
                        "notes": "testing notes",
                        "startTime": "2019-08-22 17:00:00 UTC",
                        "endTime": "2019-08-22 18:00:00 UTC",
                        "eventCode": "a1c4a4"
                    },
                    {
                        "id": 4,
                        "name": "Test Event 4",
                        "notes": "testing notes",
                        "startTime": "2019-08-23 19:00:00 UTC",
                        "endTime": "2019-08-23 20:00:00 UTC",
                        "eventCode": "e9e54f"
                    },
                    {
                        "id": 5,
                        "name": "Test Event 5",
                        "notes": "testing notes",
                        "startTime": "2019-07-18 19:00:00 UTC",
                        "endTime": "2019-07-18 20:00:00 UTC",
                        "eventCode": "136834"
                    },
                    {
                        "id": 6,
                        "name": "Test Event 6",
                        "notes": "testing notes",
                        "startTime": "2019-07-17 19:00:00 UTC",
                        "endTime": "2019-07-17 20:00:00 UTC",
                        "eventCode": "9dcfb7"
                    },
                    {
                        "id": 7,
                        "name": "Test Event 7",
                        "notes": "testing notes",
                        "startTime": "2019-07-16 19:00:00 UTC",
                        "endTime": "2019-07-16 20:00:00 UTC",
                        "eventCode": "6395f0"
                    },
                    {
                        "id": 9,
                        "name": "TODAYYY",
                        "notes": "Test today",
                        "startTime": "2019-07-23 19:30:00 UTC",
                        "endTime": "2019-07-23 21:00:00 UTC",
                        "eventCode": "b74ed5"
                    },
                    {
                        "id": 10,
                        "name": "Test Response",
                        "notes": "Testing Response",
                        "startTime": "2019-07-24 03:00:00 UTC",
                        "endTime": "2019-07-24 03:00:00 UTC",
                        "eventCode": "954b50"
                    },
                    {
                        "id": 11,
                        "name": "TODAYS EVENT",
                        "notes": "yayyy",
                        "startTime": "2019-07-24 19:00:00 UTC",
                        "endTime": "2019-07-24 20:00:00 UTC",
                        "eventCode": "2e260f"
                    }
                ]
            }
        ]
    }
  },
  mockFutureEvents: {
    "futureEvents": [
      {
          "id": 2,
          "name": "Test Event 2",
          "notes": "testing notes",
          "startTime": "2019-08-21 17:00:00 UTC",
          "endTime": "2019-08-21 18:00:00 UTC"
      },
      {
          "id": 3,
          "name": "Test Event 3",
          "notes": "testing notes",
          "startTime": "2019-08-22 17:00:00 UTC",
          "endTime": "2019-08-22 18:00:00 UTC"
      }
    ]
  },
  mockPastEvents: {
    "pastEvents": [
        {
            "id": 11,
            "name": "TODAYS EVENT",
            "notes": "yayyy",
            "startTime": "2019-07-24 19:00:00 UTC",
            "endTime": "2019-07-24 20:00:00 UTC"
        },
        {
            "id": 10,
            "name": "Test Response",
            "notes": "Testing Response",
            "startTime": "2019-07-24 03:00:00 UTC",
            "endTime": "2019-07-24 03:00:00 UTC"
        }
      ]
  },
  mockVenue: {
    allVenues: [
      {
        "id": 1,
        "name": "Venue One",
        "notes": "Here be Dragons",
        "email": "example@venue.com",
        "webUrl": "www.venueone.com",
        "addresses": [
          {
              "street1": "234 Main St.",
              "street2": "Apt 456",
              "city": "denver",
              "state": "CO",
              "zip": "80128"
          }
        ],
      }
    ]
  },
  mockStudentAttendance: {
      "id": 6,
      "username": "student1",
      "name": "Student One",
      "guardianId": {
          "name": "Matt Weiss",
          "username": "Duce",
          "email": "weiss.matt@outlook.com",
          "phoneNumber": "5555555555"
      }
  }

};

export default mockData;
