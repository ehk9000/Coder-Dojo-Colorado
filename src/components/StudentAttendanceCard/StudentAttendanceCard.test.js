import React from 'react';
import { mount } from 'enzyme';
import StudentAttendanceCard from './StudentAttendanceCard';
import mockData from '../../utils/mockData';

describe('StudentAttendanceCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<StudentAttendanceCard student={mockData.mockStudentAttendance} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render guardian info on mouseenter', () => {
    wrapper.find('StudentAttendanceCard div').simulate('mouseEnter');
    const result = wrapper.find('.guardian-header').props().className;
    expect(result).toEqual('guardian-header');
  });

  it('should render guardian info on hover', () => {
    wrapper.find('StudentAttendanceCard div').simulate('mouseLeave');
    expect(wrapper.exists('.guardian-details')).toEqual(false);
  });
});
