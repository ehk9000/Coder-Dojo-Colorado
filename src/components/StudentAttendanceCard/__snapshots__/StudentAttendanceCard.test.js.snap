// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentAttendanceCard should match snapshot 1`] = `
<StudentAttendanceCard
  student={
    Object {
      "guardianId": Object {
        "email": "weiss.matt@outlook.com",
        "name": "Matt Weiss",
        "phoneNumber": "5555555555",
        "username": "Duce",
      },
      "id": 6,
      "name": "Student One",
      "username": "student1",
    }
  }
>
  <article
    className="StudentAttendanceCard"
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <h4>
        Student One
         : 
        student1
      </h4>
    </div>
  </article>
</StudentAttendanceCard>
`;
